examples/face_recognition_11/face_recognition_11/evaluator.py

```diff
@@ -0,0 +1,182 @@
+        n_imposter_pairs=n_imposter_pairs,
+        n_fm=n_fm,
+        fmr=n_fm / n_imposter_pairs,
+        n_fnm=n_fnm,
+        fnmr=n_fnm / n_genuine_pairs,
+    )
+
+
+def compute_test_case_plots(ground_truths: List[GroundTruth], inferences: List[Inference]) -> Optional[List[Plot]]:
+    FMR_lower = -6
+    FMR_upper = -1
+    baseline_fmr_x = list(np.logspace(FMR_lower, FMR_upper, 50))
+
+    thresholds = [compute_threshold(zip(ground_truths, inferences), fmr) for fmr in baseline_fmr_x]
+
+    n_genuine_pairs = np.sum([gt.is_same for gt in ground_truths])
+    n_imposter_pairs = np.sum([not gt.is_same for gt in ground_truths])
+
+    fnmr_y = list()
+    fmr_y = list()
+
+    for threshold in thresholds:
+        tsm_for_one_threshold = [compute_per_sample(gt, inf, threshold) for gt, inf in zip(ground_truths, inferences)]
+        n_fm = np.sum([metric.is_false_match and not metric.failure_to_enroll for metric in tsm_for_one_threshold])
+        n_fnm = np.sum(
+            [metric.is_false_non_match and not metric.failure_to_enroll for metric in tsm_for_one_threshold],
+        )
+        fnmr_y.append(n_fnm / n_genuine_pairs)
+        fmr_y.append(n_fm / n_imposter_pairs)
+
+    curve_test_case_fnmr = CurvePlot(
+        title="Test Case FNMR vs. Baseline FMR",
+        x_label="Baseline False Match Rate",
+        y_label="Test Case False Non-Match Rate (%)",
+        x_config=AxisConfig(type="log"),
+        curves=[Curve(x=baseline_fmr_x, y=fnmr_y)],
+    )
+
+    curve_test_case_fmr = CurvePlot(
+        title="Test Case FMR vs. Baseline FMR",
+        x_label="Baseline False Match Rate",
+        y_label="Test Case False Match Rate",
+        x_config=AxisConfig(type="log"),
+        y_config=AxisConfig(type="log"),
+        curves=[Curve(x=baseline_fmr_x, y=fmr_y)],
+    )
+
+    return [curve_test_case_fnmr, curve_test_case_fmr]
+
+
+def compute_test_suite_metrics():
+    return
+
+
+def evaluate_face_recognition_11(
+    test_samples: List[TestSample],
+    ground_truths: List[GroundTruth],
+    inferences: List[Inference],
+    test_cases: TestCases,
+    configuration: FMRConfiguration,
+) -> EvaluationResults:
+    threshold = compute_threshold(zip(ground_truths, inferences), configuration.false_match_rate)
+
+    # compute per-sample metrics for each test sample
+    test_sample_metrics = [compute_per_sample(gt, inf, threshold) for gt, inf in zip(ground_truths, inferences)]
+
+    # compute aggregate metrics across all test cases using `test_cases.iter(...)`
+    all_test_case_metrics: List[Tuple[TestCase, TestCaseMetrics]] = []
+    all_test_case_plots: List[Tuple[TestCase, List[Plot]]] = []
+    for test_case, ts, gt, inf, tsm in test_cases.iter(test_samples, ground_truths, inferences, test_sample_metrics):
+        all_test_case_metrics.append((test_case, compute_test_case_metrics(ts, gt, tsm)))
+        all_test_case_plots.append((test_case, compute_test_case_plots(gt, inf)))
+
+    # TODO: ADD test suite metrics
```

Comment:
I'd delete all unused code

Reply:
This is for the next PR within enhancements